src/main/java/com/emc/mongoose/storage/driver/pravega/kvs/PravegaKVSDriver.java

```diff
@@ -222,59 +344,228 @@ protected boolean submit(O op) throws IllegalStateException {
+        try {
+            val kvpValueSize = op.item().size();
+            if (kvpValueSize > MAX_KVP_VALUE) {
+                Loggers.ERR.warn("{}: KVP value size cannot exceed 1016KB ", stepId);
+                completeOperation(op, FAIL_IO);
+            } else if (kvpValueSize < 0) {
+                completeOperation(op, FAIL_IO);
+            } else {
+                if (concurrencyThrottle.tryAcquire()) {
+                    op.startRequest();
+                    val kvtPutFuture = kvt.put(null, op.item().name(), op.item()); // first parameter is key family
+                    try {
+                        op.finishRequest();
+                    } catch (final IllegalStateException ignored) {
+                    }
+                    kvtPutFuture.handle((version, thrown) -> handlePutFuture(op, thrown, kvpValueSize));
+                } else {
+                    return false;
+                }
+            }
+        } catch (final IOException e) {
+            throw new AssertionError(e);
+        }
+        return true;
+    }
+
+    protected Object handlePutFuture(final O op, final Throwable thrown, final long transferSize) {
+        try {
+            if (null != thrown) {
+                completeOperation(op, FAIL_UNKNOWN);
+            }
+                op.startResponse();
+                op.finishResponse();
+                op.countBytesDone(transferSize);
+                completeOperation(op, SUCC);
+        } finally {
+            concurrencyThrottle.release();
+        }
+        return null;
+    }
+
+
+    private boolean submitRead(final O kvpOp) {
+        try {
+            val nodeAddr = kvpOp.nodeAddr();
+            val endpointUri = endpointCache.computeIfAbsent(nodeAddr, this::createEndpointUri);
+            val clientConfig = clientConfigCache.computeIfAbsent(endpointUri, this::createClientConfig);
+            val kvtName = extractKVTName(kvpOp.srcPath());
+            // create the kvt factory create function if necessary
+            val kvtFactoryCreateFunc = kvtFactoryCreateFuncCache.computeIfAbsent(
+                clientConfig,
+                KVTFactoryCreateFunctionImpl::new);
+            // create the kvt factory if necessary
+            val kvtFactory = kvtFactoryCache.computeIfAbsent(scopeName, kvtFactoryCreateFunc);
+            val kvtClientCreateFunc = kvtClientCreateFuncCache.computeIfAbsent(kvtFactory, KVTClientCreateFunctionImpl::new);
+            KeyValueTable<String, I> kvTable = kvtCache.computeIfAbsent(kvtName, kvtClientCreateFunc);
+
+            // isn't used currently
+            val key = kvpOp.item().name();
+            val family = extractKVFName(key);
+
+            if (concurrencyThrottle.tryAcquire()) {
+                kvpOp.startRequest();
+                // read by key
+                val tableEntryFuture = kvTable.get(/*family*/null, /*key*/ key);
+                val kvp = tableEntryFuture.get(controlApiTimeoutMillis, MILLISECONDS);
```

Comment:
I do realize it's most likely just for dev purposes and convenient debug, but just to be clear: you'll change get future to a future handler, right?